arwen/host/execution.go

```diff
@@ -95,14 +91,20 @@ func (host *vmHost) doRunSmartContractCall(input *vmcommon.ContractCallInput) (v
 		return vmOutput
 	}
 
-	gasForExecution, err := metering.DeductInitialGasForExecution(input, contract)
+	vmInput := runtime.GetVMInput()
+
+	err = metering.DeductInitialGasForExecution(contract)
+	if err != nil {
+		returnCode = vmcommon.OutOfGas
+		return vmOutput
+	}
+
+	err = metering.DeductAndLockGasIfAsyncStep()
```

Comment:
make the deductandLockGasIfAsyncStep inside the DeductInitialGasForExecution method.

Reply:
Done.